universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -0,0 +1,94 @@
+package modules;
+
+import static play.mvc.Results.forbidden;
+import static play.mvc.Results.unauthorized;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import org.pac4j.core.authorization.authorizer.RequireAllRolesAuthorizer;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.http.client.direct.DirectBasicAuthClient;
+import org.pac4j.http.client.indirect.FormClient;
+import org.pac4j.http.credentials.authenticator.test.SimpleTestUsernamePasswordAuthenticator;
+import org.pac4j.play.CallbackController;
+import org.pac4j.play.LogoutController;
+import org.pac4j.play.http.PlayHttpActionAdapter;
+import org.pac4j.play.store.PlayCookieSessionStore;
+import play.Environment;
+
+public class SecurityModule extends AbstractModule {
+
+  private final com.typesafe.config.Config configuration;
+  private final String baseUrl;
+  private static final String DEV_BASE_URL = "http://localhost:9000";
+
+  public SecurityModule(Environment environment, com.typesafe.config.Config configuration) {
+    this.configuration = configuration;
```

Comment:
Suggestion (blocking): please use `com.google.common.base.Preconditions.checkNotNull` on constructor args that can't be `null`.

Reply:
Done.

Reply:
thanks! these can be inline since `checkNotNull` returns its not null arg
```java
this.configuration = checkNotNull(configuration);
etc
```

Reply:
Done, that's better, thanks.